indexer/services/socks/src/lib/on-message.ts

```diff
@@ -0,0 +1,112 @@
+import { Worker } from 'worker_threads';
+
+import { InfoObject, logger, stats } from '@dydxprotocol-indexer/base';
+import { updateOnMessageFunction } from '@dydxprotocol-indexer/kafka';
+import { KafkaMessage } from 'kafkajs';
+
+import config from '../config';
+import { getChannels } from '../helpers/from-kafka-helpers';
+import { Channel, MessageToForward } from '../types';
+import { Index } from '../websocket';
+import { MessageForwarder } from './message-forwarder';
+import { Subscriptions } from './subscription';
+
+const NUM_WORKERS = 4;
+const getMessagesToForwardWorkers: Worker[] = [];
+const forwardMessageWorkers: Worker[] = [];
+
+for (let i = 0; i < NUM_WORKERS; i++) {
+  getMessagesToForwardWorkers.push(new Worker('./build/src/lib/workers/getMessagesToForwardWorker.js'));
+  forwardMessageWorkers.push(new Worker('./build/src/lib/workers/forwardMessageWorker.js'));
+}
+
+function getRandomWorker(workers: Worker[]): Worker {
+  const randomIndex = Math.floor(Math.random() * workers.length);
+  return workers[randomIndex];
+}
```

Comment:
fine for a proof-of-concept but might just use some worker-pool type thing. Have seen this one mentioned
https://www.npmjs.com/package/workerpool